modules/record_collection/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "cloudflare_record" "this" {
+  count = length(var.values)
```

Comment:
I wonder if this expression won't cause issues when parsing a list of ordered items, should the list change. I recall that this used to be a problem in the past. Surely using a `for_each` on a set was considered here, no?